File: src/frontend/libs/reservation-api/src/lib/models/drivers-license-details.dto.ts
/**
 * Driver's license details for requests (e.g., creating reservations)
 * Matches backend DriversLicenseDetailsDto structure
 */
import type { ISODateString, CountryCode } from "@orange-car-rental/shared";
import type { LicenseNumber } from "./license-number.type";

export type DriversLicenseDetails = {
  readonly licenseNumber: LicenseNumber;
  readonly licenseIssueCountry: CountryCode;
  readonly licenseIssueDate: ISODateString;
  readonly licenseExpiryDate: ISODateString;
};

/**
 * Driver's license information from customer profile response
 * Matches backend DriversLicenseDto structure with computed fields
 */
export type DriversLicenseInfo = {
  readonly licenseNumber: LicenseNumber;
  readonly issueCountry: CountryCode;
  readonly issueDate: ISODateString;
  readonly expiryDate: ISODateString;
  readonly isValid: boolean;
  readonly isEuLicense: boolean;
  readonly daysUntilExpiry: number;
};

File: src/backend/Services/Reservations/OrangeCarRental.Reservations.Api/Shared/DriversLicenseDetailsDto.cs
namespace SmartSolutionsLab.OrangeCarRental.Reservations.Api.Shared;

/// <summary>
///     Driver's license details.
/// </summary>
public sealed record DriversLicenseDetailsDto(
    string LicenseNumber,
    string LicenseIssueCountry,
    DateOnly LicenseIssueDate,
    DateOnly LicenseExpiryDate);
